src/workflows/client/client.py

```diff
@@ -0,0 +1,267 @@
+        """
+        Run the workflow and wait until completion.
+
+        Args:
+            start_event (Union[StartEvent, dict[str, Any], None]): start event class or dictionary representation (optional, defaults to None and get passed as an empty dictionary if not provided).
+            context: Context or serialized representation of it (optional, defaults to None if not provided)
+            **kwargs: Any number of keyword arguments that would be passed on as additional keyword arguments to the workflow.
+
+        Returns:
+            Any: Result of the workflow
+        """
+        if isinstance(start_event, StartEvent):
+            try:
+                start_event = start_event.model_dump()
+            except Exception as e:
+                raise ValueError(
+                    f"Impossible to serialize the start event because of: {e}"
+                )
+        if isinstance(context, Context):
+            try:
+                context = context.to_dict()
+            except Exception as e:
+                raise ValueError(f"Impossible to serialize the context because of: {e}")
+        request_body = {
+            "start_event": start_event or {},
+            "context": context or {},
+            "additional_kwargs": kwargs,
+        }
+        async with self._get_client() as client:
+            response = await client.post(
+                f"/workflows/{workflow_name}/run", json=request_body
+            )
+
+            response.raise_for_status()
+
+            return response.json()["result"]
+
+    async def run_workflow_nowait(
+        self,
+        workflow_name: str,
+        start_event: Union[StartEvent, dict[str, Any], None] = None,
+        context: Union[Context, dict[str, Any], None] = None,
+        **kwargs: Any,
+    ) -> dict[str, Any]:
+        """
+        Run the workflow in the background.
+
+        Args:
+            start_event (Union[StartEvent, dict[str, Any], None]): start event class or dictionary representation (optional, defaults to None and get passed as an empty dictionary if not provided).
+            context: Context or serialized representation of it (optional, defaults to None if not provided)
+            **kwargs: Any number of keyword arguments that would be passed on as additional keyword arguments to the workflow.
+
+        Returns:
+            dict[str, Any]: JSON representation of the handler running the workflow
+        """
+        if isinstance(start_event, StartEvent):
+            try:
+                start_event = start_event.model_dump()
+            except Exception as e:
+                raise ValueError(
+                    f"Impossible to serialize the start event because of: {e}"
+                )
+        if isinstance(context, Context):
+            try:
+                context = context.to_dict()
+            except Exception as e:
+                raise ValueError(f"Impossible to serialize the context because of: {e}")
+        request_body = {
```

Comment:
nit: I think we should build out some shared `TypedDict` models between client/server to lightly type things and make type discrepancies obvious